src/Models/Model.php

```diff
@@ -2,11 +2,22 @@
 
 namespace GloBee\PaymentApi\Models;
 
+use GloBee\PaymentApi\Exceptions\UnknownPropertyException;
+
 abstract class Model
 {
-    use PropertyTrait;
-
-    protected $properties = [];
+    /**
+     * @param $name
+     *
+     * @return mixed
+     * @throws UnknownPropertyException
+     */
+    public function __get($name)
+    {
+        if (property_exists($this, $name)) {
+            return $this->{$name};
+        }
 
-    protected $readonlyProperties = [];
+        throw new UnknownPropertyException($name);
```

Comment:
Lets use DomainException
```suggestion
        throw new DomainException("Cannot set unknown property named {$name}");
```